engine/withdraw.go

```diff
@@ -0,0 +1,222 @@
+package engine
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
+	withdrawDataStore "github.com/thrasher-corp/gocryptotrader/database/repository/withdraw"
+	"github.com/thrasher-corp/gocryptotrader/gctrpc"
+	"github.com/thrasher-corp/gocryptotrader/log"
+	"github.com/thrasher-corp/gocryptotrader/portfolio/withdraw"
+)
+
+const (
+	// ErrWithdrawRequestNotFound message to display when no record is found
+	ErrWithdrawRequestNotFound = "%v not found"
+	// ErrRequestCannotbeNil message to display when request is nil
+	ErrRequestCannotbeNil = "request cannot be nil"
+	// StatusError const for for "error" string
+	StatusError = "error"
+)
+
+// SubmitWithdrawal preforms validation and submits a new withdraw request to exchange
+func SubmitWithdrawal(exchName string, req *withdraw.Request) (*withdraw.Response, error) {
+	if req == nil {
+		return nil, errors.New(ErrRequestCannotbeNil)
+	}
+
+	var err error
+	var ret *withdraw.ExchangeResponse
+	if req.Exchange == "" {
+		req.Exchange = exchName
+	}
+
+	err = withdraw.Validate(req)
+	if err != nil {
+		return nil, err
+	}
+
+	exch := GetExchangeByName(exchName)
+	if exch == nil {
+		return nil, ErrExchangeNotFound
+	}
+
+	resp := &withdraw.Response{
+		Exchange: &withdraw.ExchangeResponse{
+			Name: exchName,
+		},
+		RequestDetails: req,
+	}
+
+	if Bot.Settings.EnableDryRun {
+		log.Warnln(log.Global, "Dry run enabled, no withdrawal request will be submitted or have an event created")
+		resp.ID = withdraw.DryRunID
+		resp.Exchange.Status = "dryrun"
+		resp.Exchange.ID = withdraw.DryRunID.String()
+	} else {
+		if req.Type == withdraw.Fiat {
+			ret, err = exch.WithdrawFiatFunds(req)
+			if err != nil {
+				resp.Exchange.ID = StatusError
+				resp.Exchange.Status = err.Error()
+			}
+			resp.Exchange.Status = ret.Status
+			resp.Exchange.ID = ret.ID
+		} else if req.Type == withdraw.Crypto {
+			ret, err = exch.WithdrawCryptocurrencyFunds(req)
+			if err != nil {
+				resp.Exchange.ID = StatusError
+				resp.Exchange.Status = err.Error()
+			} else {
+				resp.Exchange.Status = ret.Status
+				resp.Exchange.ID = ret.ID
+			}
+		}
+		withdrawDataStore.Event(resp)
+	}
+
```

Comment:
Pettiest of nits, but please remove line. The err check is all alone when it should be with its err making code

Reply:
i guess i could do this i mean its such a hassle but :stuck_out_tongue: